test_util/aws.py

```diff
@@ -55,7 +55,7 @@ def create_stack(self, name, template_url, user_parameters, deploy_timeout=60):
         log.info('Requesting AWS CloudFormation...')
         cf_parameters = []
         for k, v in user_parameters.items():
-            cf_parameters.append({'ParameterKey': k, 'ParameterValue': v})
+            cf_parameters.append({'ParameterKey': k, 'ParameterValue': str(v)})
```

Comment:
I tried rewriting this with a function that pulls the template in advance, looks in the key section, checks that they keys the user provided are there, and then looks at the parameter type in the template and casts the parameter as that type if necessary. However, AWS template parameter types do NOT apply to python. Every parameters MUST be a string. This is especially confusing given that Azure does the opposite, enforcing parameter types in python to be consistent with the parameter type in the template

For example. here we see a boto validation error because I passed two values as floats being as the parameter type in the template is "Number" and the dcos say "Numbers can be integers or floats". 
https://teamcity.mesosphere.io/viewLog.html?buildId=480558&buildTypeId=ClosedSource_Dcos_IntegrationTests_CloudIntegrationTests_ProvisioningTestAwsDe_2&tab=buildLog#_focus=164

So, no matter what parameters you are passing, they need to be string here, thus, this is the best place to double check that.